src/tyr/route_serializer_osrm.cc

```diff
@@ -549,6 +549,14 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
       edges.emplace_back(((prior_heading + 180) % 360), entry, true, false);
     }
 
+    if (i > 0 && !arrive_maneuver) {
+      bool entry = (arrive_maneuver) ? true : false;
```

Comment:
1. `arrive_maneuver` is always equal to `false` in this branch so this variable/ternary operator is redundant
2. The entry is set to false if the intersecting edge is not routable in that direction. I'm not sure that is should be always false..